ports/geckolib/gecko_bindings/sugar/refptr.rs

```diff
@@ -0,0 +1,91 @@
+use std::marker::PhantomData;
+use std::mem::{forget, transmute};
+use std::ptr::{read, null};
+use std::sync::Arc;
+
+/// Helper trait for conversions between FFI Strong/Borrowed types and Arcs
+pub unsafe trait RefCounted where Self: Sized {
```

Comment:
I feel like this is less about being reference counted, and more about being a type T which has exposed Arc<T>s to Gecko. Perhaps HasArcFFI or similar would be a better name?